Cesium3DTilesSelection/src/TilesetJsonLoader.cpp

```diff
@@ -548,16 +553,21 @@ std::optional<Tile> parseTileJsonRecursively(
   const auto contentIt = tileJson.FindMember("content");
   bool hasContentMember =
       (contentIt != tileJson.MemberEnd()) && (contentIt->value.IsObject());
-  const char* contentUri = nullptr;
+
+  std::optional<Content> maybeContent;
   if (hasContentMember) {
-    auto uriIt = contentIt->value.FindMember("uri");
-    if (uriIt == contentIt->value.MemberEnd() || !uriIt->value.IsString()) {
-      uriIt = contentIt->value.FindMember("url");
-    }
+    ContentReader contentReader;
+    auto contentResult = contentReader.readFromJson(contentIt->value);
+    maybeContent = std::move(contentResult.value);
+  }
 
-    if (uriIt != contentIt->value.MemberEnd() && uriIt->value.IsString()) {
-      contentUri = uriIt->value.GetString();
-    }
+  const char* contentUri = nullptr;
+  if (maybeContent && !maybeContent->uri.empty()) {
+    contentUri = maybeContent->uri.c_str();
+  } else if (maybeContent && maybeContent->unknownProperties.contains("url")) {
+    std::string url =
+        maybeContent->unknownProperties["url"].getStringOrDefault({});
+    contentUri = !url.empty() ? url.c_str() : nullptr;
```

Comment:
This will be a use-after-free error. The memory pointed to by `contentUri` will be freed when `url` is destroyed.

I'd suggest something like this:

```cpp
if (maybeContent && maybeContent->uri.empty()) {
  auto it = maybeContent->unknownProperties.find("url");
  if (it != maybeContent->unknownProperties.end()) {
    maybeContent->uri = it->getStringOrDefault({});
  }
}
```

Reply:
whoops 🤦 thanks for the catch!